src/system.rs

```diff
@@ -42,12 +42,127 @@ impl PendingTypes {
+                        // we haven't yet resolved this import.
+                        return Ok(false);
+                    }
+                }
+                IslImport::TypeAlias(isl_import) => {
+                    // if import has a specified type with an alias then renamed that type to the given alias and add it
+                    if type_store
+                        .imported_type_ids_by_name
+                        .contains_key(isl_import.alias().as_ref().unwrap())
+                    {
+                        // if the type_store already has the import in it then return true (i.e. TypeAlias has already been imported)
+                        return Ok(true);
+                    } else if let Some(named_type_def) =
+                        self.get_type_by_name_for_import(isl_import.type_name(), type_store)
+                    {
+                        let aliased_type_def = named_type_def?
+                            .with_name(isl_import.alias().as_ref().unwrap().to_owned());
+                        type_store.add_import_type(isl_import.alias().as_ref(), aliased_type_def);
+                        self.update_type_store_with_all_import_types(
+                            Some(isl_import.type_name()),
+                            type_store,
+                        )?;
+                    } else {
+                        // if the named_type_def appears as None then it means we haven't reached
+                        // the named type to be imported yet hence we return with false pointing
+                        // we haven't yet resolved this import.
+                        return Ok(false);
+                    }
+                }
+            }
+        } else {
+            // if load_isl_import is None i.e. it is the root schema, then update type_store with all the types inside this PendingTypes
+            self.update_type_store_with_all_types(type_store)?;
+        }
+        self.types_by_id.clear();
+        self.ids_by_name.clear();
+        Ok(true)
+    }
+
+    // helper method get named type for given import_type_name from this PendingTypes
+    // this return type will be used by update_type_store method to then update type_store with this named type as import
+    fn get_type_by_name_for_import(
+        &self,
+        import_type_name: &str,
+        type_store: &mut TypeStore,
+    ) -> Option<IonSchemaResult<TypeDefinitionImpl>> {
+        let type_def = self
+            .get_type_id_by_name(import_type_name, type_store)
+            .and_then(|id| self.types_by_id[id].to_owned());
+
+        match type_def {
+            None => None,
+            Some(type_def) => match type_def {
+                TypeDefinition::Named(named_type_def) => Some(Ok(named_type_def)),
+                TypeDefinition::Anonymous(_) => Some(unresolvable_schema_error(format!(
+                    "Unable to load import type: {} for schema",
+                    import_type_name
+                ))),
```

Comment:
If I'm reading this right, this would be the situation in which the user requested a type by name and found an anonymous type in the type store? If that's accurate, I think this is use case for [`unreachable!`](https://doc.rust-lang.org/std/macro.unreachable.html#). We don't expect this to ever happen and there's nothing a user could do with the error if we handed it to them.